pkg/kgo/producer.go

```diff
@@ -364,6 +364,8 @@ func (cl *Client) produce(
 ) {
 	if ctx == nil {
 		ctx = context.Background()
+	} else {
+		r.Context = ctx
```

Comment:
I think we want to set the context unconditionally to whatever `ctx`, how about dropping the `else`?